### PR TITLE
Kontrol s4 mappings LED updates and autoslip mode

### PR DESCRIPTION
**Pull request by fayaaz**
Summary:

- Auto Slip Mode added:
  - Press Play (in the loop recorder) + Flux
  - Remix LED lights up
  - Jog wheel scratches will start slip mode on wheel press and exit on touch up
  - Entering a loop will start slip mode, on loop exit slip mode also exits
  - Hotcues on touchdown will start slip mode, on touchup exit
- Add range for speed splider - 8, 16, 24, 50, 90 : Shift + flux/reset buttons
- LED colours slightly changed to match LateNight
- Minor code changes

### CODE REVIEW

File: res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js [hunk 1]
```diff
@@ -1598,6 +1645,75 @@ TraktorS4MK2.loopSizeSet = function(group) {
...
+      var prev_range = TraktorS4MK2.controller.tempoRange[field.group];
+      print('Previous range ' + prev_range);
+      if (prev_range >= TraktorS4MK2.controller.tempoRangeList.length - 1){
+        prev_range = TraktorS4MK2.controller.tempoRangeList.length - 1;
+        TraktorS4MK2.controller.tempoRange[field.group] = prev_range;
+      } else {
+        var new_range = prev_range + 1;
+        var range_value = TraktorS4MK2.controller.tempoRangeList[new_range] / 100;
+        TraktorS4MK2.controller.tempoRange[field.group] = new_range;
+        engine.setValue(field.group, "rateRange", range_value);
```
Comment (Be-ing) [inline res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js]: The variable names in this function are confusing like the `TraktorS4Mk2.slipHandler` function above.

**Review by Be-ing — CHANGES_REQUESTED** [review]
The UX changes seem fine. Please document them on the [wiki page](https://mixxx.org/wiki/doku.php/native_instruments_traktor_kontrol_s4_mk2). I have made some comments below to help make the code easier to read.

The slip auto mode [should be handled in Mixxx instead](https://bugs.launchpad.net/mixxx/+bug/1475303). It is okay to have it implemented in this mapping for now to work around that, but it should be easy to map with any controller. Would you be interested in working on that?

Have you made any progress figuring out why this mapping is not working on Windows?

**Comment by daschuer** [comment]
Can we move this to 2.1.x and release 2.1.1 this week?

**Comment by Be-ing** [comment]
This is already assigned to the 2.1.1 milestone.

**Comment by daschuer** [comment]
This is not ready. I like to shift it to the next point release.

**Comment by Be-ing** [comment]
ping @fayaaz, could you take care of the review comments above?

**Comment by Holzhaus** [comment]
Ping @fayaaz

**Pull request closed without merging**